### DISCUSSION TITLE
I have added GT911 touch screen support for GUISlice

### DISCUSSION THREAD

**Discussion by DeanK2022** [ImpulseAdventure/GUIslice#500] [Ideas]
The repo is https://github.com/DeanK2022/GUISlice, hope that is OK.

**Comment by Pconti31** [comment]
@DeanK2022 I would suggest that If you want the code included you simply need to issue a pull request.
Note that Calvin is busy on a work project so it could be a while before he can look over your code.

Paul--

**Reply by DeanK2022** [reply]
Thanks Paul.  
There's a pretty big difference between reactive and touch capacitive events I think, there is only a "pressed" state for capacitive screens.  GUISlice seems to need a "press" and "release" event in the touchscreen driver, and the changes I made were butchered rather than coded as a result.

To make a decent pull request would likely need a fair bit of re-architecting and that is not something I would feel confident in doing well as there's big chunks of the GUISlice codebase I haven't looked at.  The autogenerated .h and .ino needed fixing up (see BladeOMatic repo) and there seems to be some confusion between gslc_teTouch and gslc_teInputRawEvent classes and enumerations (although that is likely because I don't know the code very well).

If someone with a better knowledge of GUISlice wants to take my code and integrate it properly then search my repo for DRV_TOUCH_TFT_GT911 and see the changes in GUISlice.c as well.

Thanks to you and Calvin for GUISlice, nicely done gents!

**Comment by Pconti31** [comment]
@DeanK2022 ok, you know best. But guislice already supports cap driver ft6236 (see ard-shld-eastrising-35-il9488-cap.h) which is similar to the gt911. The fancy gestures are not supported but basic touch works.

**Reply by DeanK2022** [reply]
Thanks @Pconti31 
>you know best 
Heh, I don't, that is why I am shying away from filing a PR! :)
I will take a look at ard-shld-eastrising-35-il9488-cap.h and see if there's an easy way to mimic that behaviour on the gt911 side.  If it solves the autogenerate code issues I have as well then that'd be excellent and I could then potentially put a PR together.
Thanks for your help!

**Comment by RSanchez888** [comment]
Dear Deank,
I am using your library GUIslice to build our customized touchscreens for a capacitive ILI9488 (GT911). Likewise, I am using the GUI slice builder software to design the touchscreens and generate the code. Unfortunately, the code does not compile  on the arduino IDE with the following errors:
1) _class GT911 has no member named "i2cAddr"_ 
2) _class GT911 has no member named "touchcount"_ 
3) _class GT911 has no member named "readPoints"_ 

We have tested one of the examples in your library and the same errors show up.
Do you think these errors might be related with the version of the GT911 library? Which version are you using?
If not, could you please suggest any hint to solve the issues?

Thanks a lot in advance!

**Reply by DeanK2022** [reply]
You are going to have to look at the whole projects, see BladeOMatic which pulls in all the custom stuff I had to do to get it working.  It is not standalone I'm afraid.

**Comment by Pconti31** [comment]
@RSanchez888 
@DeanK2022 
From the above messages I assume he has his own driver repro [DeanK2022/GT911](https://github.com/DeanK2022/GT911) and that even so you need to edit the GUIsliceBuilder output to correctly compile the auto-generated code.
You should look at his example program [DeanK2022/BladeOMatic](https://github.com/DeanK2022/BladeOMatic)

I don't own one of these so I can't really be of much further help.
Paul--

**Reply by RSanchez888** [reply]
Dear Paul, thanks a lot for your replies. I have downloaded the driver repro Deank2022/GT911 and the compilation issues disappeared as you suggested. The problem now is that upon uploading the autogenerated codes of the screens that I created with GUIslice Builder, my display does not show anything (white screen). To try to figure out what is the problem, I downloaded the [DeanK2022/BladeOMatic](https://github.com/DeanK2022/BladeOMatic) example with the same pins configuration but a compilation error shows up: `BladeOMatic.ino:69:3: error: 'gslc_ElemXCheckboxSetColour' was not declared in this scope gslc_ElemXCheckboxSetColour(&m_gui, m_pElemCBCheckBoxShort, GSLC_COL_BLUE);`

**Reply by RSanchez888** [reply]
By the way, I wrote an example code based on the libraries `ILI9486_SPI` and `TAMC_GT911` and I make work the TFT and touch correctly, so I assume is not due to a wiring issue. The point is that I would like to use the Deank2022 GUIslice library to make my own touch GUI.

**Reply by DeanK2022** [reply]
This is the missing function, add it into GUIslice\src\elem\XCheckbox.c

void gslc_ElemXCheckboxSetColour(gslc_tsGui* pGui, gslc_tsElemRef* pElemRef, gslc_tsColor tscColour)
{
    gslc_tsXCheckbox* pCheckbox = (gslc_tsXCheckbox*)gslc_GetXDataFromRef(pGui, pElemRef, GSLC_TYPEX_CHECKBOX, __LINE__);
    if (!pCheckbox) return;

    pCheckbox->colCheck = tscColour;
    gslc_ElemSetRedraw(pGui, pElemRef, GSLC_REDRAW_INC);

}

That code is pretty specific to the board I am using, see the write up on https://github.com/DeanK2022/BladeOMatic for details.

What exact chip/board are you using?  I had to faff a lot with pull up resistors and juggle ESP32 pins to get everything working.

**Comment by Pconti31** [comment]
@RSanchez888 You might also want to checkout [Issue 522](https://github.com/ImpulseAdventure/GUIslice/issues/522)
Paul--

**Comment by Pconti31** [comment]
@RSanchez888 I quick search of DeanK2022/GUISlice  shows that there is no routine called gslc_ElemXCheckboxSetColour so jus try and comment it out and see what happens?
